src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -20,11 +18,11 @@
 import static com.atlassian.db.replica.api.reason.Reason.REPLICA_INCONSISTENT;
 import static com.atlassian.db.replica.api.reason.Reason.RO_API_CALL;
 import static com.atlassian.db.replica.api.reason.Reason.RW_API_CALL;
-import static com.atlassian.db.replica.api.state.State.CLOSED;
-import static com.atlassian.db.replica.api.state.State.COMMITED_MAIN;
-import static com.atlassian.db.replica.api.state.State.MAIN;
-import static com.atlassian.db.replica.api.state.State.NOT_INITIALISED;
-import static com.atlassian.db.replica.api.state.State.REPLICA;
+import static com.atlassian.db.replica.internal.state.State.CLOSED;
```

Comment:
Can you describe the reasoning behind hiding  State.
It can't be concluded from the changes.

Reply:
Removed API goes thru a deprecation cycle. And deprecation should point to the replacement.
With SemVer 0.x.y we can break stuff all the time, but we should use deprecation anyway, because:
* the consumers will know how to bump
* it's self-explanatory for other maintainers
* we're on prod already, so [we should already be 1.0.0+](https://semver.org/#how-do-i-know-when-to-release-100)
* deprecation is an important muscle to train

Reply:
I'm preparing for `1.0.0` release. There's no clear use-case or user for this API because it was replaced by `RouteDecision` API (the same goal, but a different approach). It's safe to keep it out, as it gives us more options for the evolution of the API (we can bring it back, remove, or add it in a better shape after we have a real user and use-case).

Reply:
> I don't understand how deprecation will help here. We are removing it from API without providing any alternative. The goal is to unlock 1.0.0 end to release without doubtful or broken features.

The alternative is `RouteDecision`.
I am all in favor of removing doubtful features. But we already got consumers. We can deprecate now and remove in the next release.
Do you know that there are no usages of `State`? Having such extra external knowledge would explain why you see deprecation as unnecessary.

Reply:
> We can deprecate now and remove in the next release.

We can but is it worth it?
1. Deprecate
2. Release
3. Safe-rollout
4. Remove
5. Release
6. Safe-rollout

We could put this extra effort into making the lib better instead of for formality with non-semver version when the only known customer doesn't use it and have never used it.

Reply:
In SemVer, there will be no extra overhead. You add feature and deprecate OLD API at the same release.

Reply:
> 3. Safe-rollout
> 5. Release
> 6. Safe-rollout

We can skip these steps.

Could you address this question?
> Do you know that there are no usages of `State`?

Or maybe deprecation isn't worth it, because we know only one consumer and we can use IDE to proactively find the usages?

We can skip the deprecation, but we need a good reason for that. I'm digging, because I care about the decision-making, not the outcome. The end result will be the same and we will move on, but the reasoning we used here will be reused later on many other occasions. It has to be solid.

Here's my reason for skipping the deprecation: [SemVer rule 4](https://semver.org/#spec-item-4). I argued that we *should be* on 1.0.0 due to [a SemVer hint](https://semver.org/#how-do-i-know-when-to-release-100), but we *are not*. So the hint is weaker than the rule.

Reply:
> We can skip these steps.

How? 

> Do you know that there are no usages of State?

No. but It's pre `1.0.0`. It can be removed anytime.

Reply:
> How?

1. Deprecate
2. Release
3. Safe-rollout // not needed, because we can't `if` around a new class or around a `@Deprecated`, compile-time safety must be enough
4. Remove
5. Release // not needed, removing a class brings no value to the consumer, why would they bump for that?
6. Safe-rollout // not needed, because there's no release and also point 3. remains

> No. but It's pre 1.0.0. It can be removed anytime.

Correct, that's SemVer rule 4 👍